storm-core/src/ui/public/topology.html

```diff
@@ -384,8 +389,51 @@ <h2 id="topology-resources-header">Topology resources</h2>
             $('#topology-configuration [data-toggle="tooltip"]').tooltip();
             $('#topology-actions [data-toggle="tooltip"]').tooltip();
             $('#topology-visualization [data-toggle="tooltip"]').tooltip();
+
+            var lagUrl = "/api/v1/topology/"+topologyId+"/lag";
+            $.getJSON(lagUrl,function(lagResponse,status,jqXHR) {
+              if (lagResponse !== null && lagResponse !== undefined && lagResponse instanceof Array && lagResponse.length > 0) {
+                var kafkaSpoutsLagTemplate = $(template).filter("#topology-kafka-spouts-lag-template").html();
+                var spoutsErrorTemplate = $(template).filter("#topology-spouts-lag-error-template").html();
+
+                var kafkaSpoutLags = lagResponse.filter(function(ele) {return ele.spoutType === "KAFKA";});
+                var isJson = function (input) {
+                  try {
+                    JSON.parse(input);
+                  } catch (e) {
+                    return false;
+                  }
+                  return true;
+                };
+                var kafkaSpoutsValidResults = kafkaSpoutLags.filter(function (ele) {return isJson(ele.spoutLagResult);});
+                var kafkaSpoutsErrorResults = kafkaSpoutLags.filter(function (ele) {return !isJson(ele.spoutLagResult);});
+                var data = {};
+                if (kafkaSpoutsValidResults.length > 0) {
+                  data.kafkaSpoutsLagResults = [];
+                  kafkaSpoutsValidResults.forEach(function(ele) {
+                    var spoutLagResult = JSON.parse(ele.spoutLagResult);
+                    spoutLagResult.forEach(function(ele2) {
```

Comment:
We would still need the for loop as need to flatten the results out. I chose to use id in the json passed to template since the template uses id in other tables for ui. If you are fine we can leave it as is.